File: Arrays/Longest_subarray_with_sum_k/brute_sol.py
def longestSubarray(num,k):
    n = len(num)
    maxLength = 0
    for i in range(n):
        sum = 0
        for j in range(i,n):
            sum += num[j]

            if sum == k:
                maxLength = max(maxLength,j-i+1)

    return maxLength


if __name__ == "__main__":
    nums = [-1, 1, 1]
    k = 1
    length = longestSubarray(nums,k)
    print("The length of the longest subarray is:", length)

File: Arrays/Longest_subarray_with_sum_k/brute_sol.cpp
#include <bits/stdc++.h>
using namespace std;

int longestSubarray(vector<int> &nums, int k){
    int n = nums.size();
    int maxLength = 0;

    for(int i=0; i<n; i++){
        int sum = 0;
        for(int j=i; j<n; j++){
            sum += nums[j];
            if( sum == k){
                maxLength = max(maxLength,j=i+1);
            }

        }
        
    }
    return maxLength;
}

int main(){
    vector<int> a ={-1,1,1};
    int k = 1;
    int len = longestSubarray(a,k);
    cout << "The length of the longest subarray is: " << len << "\n";
    return 0;
}